### PR TITLE
Upload endpoints

### PR DESCRIPTION
**Pull request by mmcc**

### CODE REVIEW

File: src/data/data.js [hunk 1]
```diff
@@ -32,19 +32,24 @@ class Data extends Base {
     super(...params);
 
     /** @type {Metrics} */
-    this.metrics = new Metrics(this);
+    this.Metrics = new Metrics(this);
+    this.metrics = this.Metrics;
```
Comment (jsanford8) [inline src/data/data.js]: Why do you still need these? Just so this doesn't break backwards compatibility?

File: src/video/resources/uploads.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+ * */
+const buildBasePath = uploadId => `${PATH}/${uploadId}`;
+
+/**
+ * Uploads Class - Provides access to the Mux Video Uploads API
+ *
+ * @example
+ * const { Video } = new Mux(accessToken, secret);
+ *
+ * // Create an upload
```
Comment (jsanford8) [inline src/video/resources/uploads.js]: Shouldn't this be around line 38... or just removed?

File: src/video/resources/uploads.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  /**
+   * Deletes an upload
+   * @param {string} uploadId - The ID for the upload intended for deletion
+   * @returns {Promise} - Returns a resolved Promise with a response from the Mux API
+   *
+   * @example
+   * const { Video } = new Mux(accessToken, secret);
+   *
+   * // Delete an upload
+   * Video.Uploads.remove(uploadId);
```
Comment (jsanford8) [inline src/video/resources/uploads.js]: Do we use `remove` elsewhere for delete things? Like... why not `delete?

Cool if we already use remove elsewhere, just asking

File: test/integration/video/uploads.spec.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+/** @test {Uploads} */
+describe('Integration::Uploads', () => {
+  const muxClient = new Mux();
+  const { Video } = muxClient;
+  let testUpload;
+  const createdUploads = []; // These are uploads we'll clean up when it's all done.
+
+
+  before(async () => {
```
Comment (jsanford8) [inline test/integration/video/uploads.spec.js]: Why do you do this? You have a test specifically for this, and then create one directly in each test where you need to do things...

Comment (jsanford8) [reply test/integration/video/uploads.spec.js]: Except for the get call... why not just create it in the get test?

File: test/unit/data/data.spec.js [hunk 5]
```diff
@@ -27,11 +27,11 @@ describe('Unit::Data', () => {
...
-      expect(TestData.metrics).to.be.an.instanceof(Metrics);
-      expect(TestData.errors).to.be.an.instanceof(Errors);
-      expect(TestData.exports).to.be.an.instanceof(Exports);
-      expect(TestData.videoViews).to.be.an.instanceof(VideoViews);
-      expect(TestData.filters).to.be.an.instanceof(Filters);
+      expect(TestData.Metrics).to.be.an.instanceof(Metrics);
+      expect(TestData.Errors).to.be.an.instanceof(Errors);
+      expect(TestData.Exports).to.be.an.instanceof(Exports);
+      expect(TestData.VideoViews).to.be.an.instanceof(VideoViews);
+      expect(TestData.Filters).to.be.an.instanceof(Filters);
```
Comment (jsanford8) [inline test/unit/data/data.spec.js]: If we're keeping the lower-cased versions around for backwards compatibility, should we add tests to make sure those exist so we don't accidentally remove them for now?

File: test/unit/video/video.spec.js [hunk 6]
```diff
@@ -24,8 +25,10 @@ describe('Unit::Video', () => {
     it('creates a new Video instance', () => {
       const TestVideo = new Video(testApiKey, testSecret);
       expect(() => new Video(testApiKey, testSecret)).to.not.throw();
-      expect(TestVideo.assets).to.be.an.instanceof(Assets);
-      expect(TestVideo.liveStreams).to.be.an.instanceof(LiveStreams);
+      expect(TestVideo.Assets).to.be.an.instanceof(Assets);
+      expect(TestVideo.LiveStreams).to.be.an.instanceof(LiveStreams);
+      expect(TestVideo.LiveStreams).to.be.an.instanceof(LiveStreams);
+      expect(TestVideo.Uploads).to.be.an.instanceof(Uploads);
```
Comment (jsanford8) [inline test/unit/video/video.spec.js]: Same here with the lower-cased ones...

File: README.md [hunk 7]
```diff
@@ -22,7 +22,7 @@ yarn add @mux/mux-node
 ```
 
 ## Releases
-The latest **stable** release is `1.1.1`
+The latest **stable** release is `2.0.0`
```
Comment (jsanford8) [inline README.md]: Ooh if this is a major version bump... let's just kill the lower-cased versions of things?

File: src/data/data.js (hunk 1 above)
Comment (mmcc) [reply src/data/data.js]: Yeah, just so there'd be one less big change to make if you were updating from 1.x (you already have to remove all the `data.data` accessors)

File: src/video/resources/uploads.js (hunk 2 above)
Comment (mmcc) [reply src/video/resources/uploads.js]: It's just an example for using the class module, but I was just being consistent with the other modules and don't feel too strongly about it.

File: src/video/resources/uploads.js (hunk 3 above)
Comment (mmcc) [reply src/video/resources/uploads.js]: Yeah, all the other modules used `remove`.

Comment (JacquiManzi) [reply src/video/resources/uploads.js]: `delete` is a reserved word in Javascript and i believe we were using `deleteAsset` or `removeAsset` before, but there was a request to keep it to one word for consistency.

File: src/video/resources/assets.js [hunk 8]
```diff
@@ -136,8 +132,7 @@ class Assets extends Base {
    * @returns {Promise} - Returns a resolved Promise with a response from the Mux API
    *
    * @example
-   * const muxClient = new Mux(accessToken, secret);
-   * const { Video } = muxClient;
+   * const { Video }= new Mux(accessToken, secret);
```
Comment (jsanford8) [inline src/video/resources/assets.js]: you're missing a space after `{ Video }` and before the `=`

**Pull request merged**